modules/core/src/main/java/org/apache/ignite/internal/util/IgniteUtils.java

```diff
@@ -12557,4 +12559,24 @@ public static int hashToIndex(int hash, int size) {
         return safeAbs(hash % size);
     }
 
+    /**
+     * Invokes {@link ServerSocket#accept()} method on the passed server socked, working around the
+     * https://bugs.openjdk.java.net/browse/JDK-8247750 in the process.
+     *
+     * @param srvrSock Server socket.
+     * @return New socket.
+     * @throws IOException If an I/O error occurs when waiting for a connection.
+     * @see ServerSocket#accept()
+     */
+    public static Socket acceptServerSocket(ServerSocket srvrSock) throws IOException {
+        while (true) {
+            try {
+                return srvrSock.accept();
+            }
+            catch (SocketTimeoutException e) {
+                if (srvrSock.getSoTimeout() != 0)
```

Comment:
AFAIK timeout can be "-1" (according to this https://hg.openjdk.java.net/jdk/jdk/rev/6e36b8dda5ab#l2.26), maybe `> 0` will be more suitable here?